package cvmfs;

option optimize_for = LITE_RUNTIME;

enum EnumStatus {
  STATUS_OK = 0;
  STATUS_FAIL = 1;
}

enum EnumHashAlgorithm {
  HASH_SHA1 = 1;
  HASH_RIPEMD160 = 2;
  HASH_SHAKE128 = 3;
}

message Hash {
  required EnumHashAlgorithm algorithm = 1;

}

// client --> cache manager
message HandshakeRequest {
  optional string name = 1;
}

// cache manager --> client
message HandshakeReply {
  required EnumStatus status = 1;
  optional int64 client_id = 2;
}

message PutRequest {
  optional int64 client_id = 1;
}
